src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisCard.tsx

```diff
@@ -0,0 +1,724 @@
+          data {
+            name
+            downloadUrl
+            reportUrl
+            metadata {
+              labels
+            }
+            size
+            archivedTime
+          }
+        }
+      }`,
+        { connectUrl }
+      );
+    },
+    [context.api, context.api.graphql]
+  );
+
+  const handleStateErrors = React.useCallback(
+    (errorMessage: string) => {
+      setErrorMessage(errorMessage);
+      setIsError(true);
+      setIsLoading(false);
+      setUsingArchivedReport(false);
+      setUsingCachedReport(false);
+    },
+    [setErrorMessage, setIsError, setIsLoading, setUsingArchivedReport, setUsingCachedReport]
+  );
+
+  const handleLoading = React.useCallback(() => {
+    setIsLoading(true);
+    setIsError(false);
+    setUsingArchivedReport(false);
+    setUsingCachedReport(false);
+  }, [setIsLoading, setIsError, setUsingArchivedReport, setUsingCachedReport]);
+
+  const handleArchivedRecordings = React.useCallback(
+    (recordings: ArchivedRecording[]) => {
+      const freshestRecording = recordings.reduce((prev, current) =>
+        prev?.archivedTime > current?.archivedTime ? prev : current
+      );
+      addSubscription(
+        context.target.target().subscribe((target) => {
+          context.reports
+            .reportJson(freshestRecording, target.connectUrl)
+            .pipe(first())
+            .subscribe({
+              next: (report) => {
+                setUsingArchivedReport(true);
+                setReportTime(freshestRecording.archivedTime);
+                categorizeEvaluation(report);
+              },
+              error: () => {
+                handleStateErrors(INTERNAL_ERROR_MESSAGE);
+              },
+            });
+        })
+      );
+    },
+    [
+      addSubscription,
+      context.target,
+      context.reports,
+      categorizeEvaluation,
+      handleStateErrors,
+      setUsingArchivedReport,
+      setReportTime,
+    ]
+  );
+
+  const handleEmptyRecordings = React.useCallback(
+    (connectUrl: string) => {
+      const cachedReportAnalysis = context.reports.getCachedAnalysisReport(connectUrl);
+      if (cachedReportAnalysis.report.length > 0) {
+        setUsingCachedReport(true);
+        setReportTime(cachedReportAnalysis.timestamp);
+        categorizeEvaluation(cachedReportAnalysis.report);
+      } else {
+        addSubscription(
+          queryArchivedRecordings(connectUrl)
+            .pipe(
+              first(),
+              map((v) => v.data.archivedRecordings.data as ArchivedRecording[])
+            )
+            .subscribe({
+              next: (recordings) => {
+                if (recordings.length > 0) {
+                  handleArchivedRecordings(recordings);
+                } else {
+                  handleStateErrors(NO_RECORDINGS_MESSAGE);
+                }
+              },
+              error: () => {
+                handleStateErrors(NO_RECORDINGS_MESSAGE);
```

Comment:
I think we should use the error.errorMessage returned from the api request instead?